os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -208,112 +211,68 @@ static int is_text_address(unsigned long programCounter)
-	char c;
-	int word;
-	int ch_in_word;
-	int read_line;
-	int ch_in_line;
-
-	if (recursive_abort) {
-		/* If there is a crash in file system, below operation would
-		   lead to recursive abort, as this api will be called inside
-		   abort handler and tries to access the file system api.
-		   To avoid recursive abort, just return from here if we are
-		   already in abort mode.
-		 */
-		return -1;
+	if (up_symtab_flashdrv_init() == OK) {
+		symtab_init = true;
 	}
+}
 
-	pFile = fopen("/rom/System.map", "r");
+static unsigned long getaddr(int index)
+{
+	unsigned long addr;
+	up_symtab_flashdrv_read(&addr, (index - 1) * 4 + 4, 4);
+	return addr;
+}
 
-	/* Check if file exists */
-	if (pFile == NULL) {
-		lldbg("Could not open file: /rom/System.map\n");
-		return -1;
-	}
-	// obtain file size:
-	fseek(pFile, 0, SEEK_END);
-	last = ftell(pFile);
+static void getname(int ind, char *buffer, size_t buflen)
+{
+	unsigned long sym_offlen;
+	unsigned long nextsym_offlen;
+	unsigned long symlen;
+
+	memset(buffer, '\0', buflen);
+	up_symtab_flashdrv_read(&sym_offlen, (symbolnum * 4) + ((ind - 1) * 4) + 4, 4);
+	up_symtab_flashdrv_read(&nextsym_offlen, (symbolnum * 4) + (ind * 4) + 4, 4);
+	symlen = nextsym_offlen - sym_offlen;
+	up_symtab_flashdrv_read(buffer, symbolnum * 8 + sym_offlen + 4, symlen);
+}
 
-	rewind(pFile);
-	mid = (first + last) / 2;
-	while (first <= last) {
-		fseek(pFile, mid, SEEK_SET);
+static void getsymbolnum(void)
+{
+	up_symtab_flashdrv_read(&symbolnum, 0, 4);
+}
 
-		/* If the file pointer is in the mid of the line, make sure
-		 * it's been properly moved to start of next line
-		 */
-		for (c = getc(pFile); c != '\n'; c = getc(pFile)) {
-			if (c == EOF) {
-				lldbg("Reached end of file and couldn't find symbol\n");
-				fclose(pFile);
-				return -1;
-			}
-		}
+int get_symbol(unsigned long search_addr, char *buffer, size_t buflen)
+{
+	int first;
+	int last;
+	int mid;
+	unsigned long addr;
+	unsigned long addr_next;
 
-		word = 0;
-		// Read 2 lines and Split the string as words
-		for (read_line = 0; read_line < 2; read_line++) {
-			fgets(line, 128, pFile);
-			ch_in_line = 0;
-			ch_in_word = 0;
-			while (line[ch_in_line] != '\0') {
-				if (line[ch_in_line] != ' ') {
-					data[word][ch_in_word++] = line[ch_in_line];
-				} else {
-					data[word][ch_in_word] = '\0';
-					word++;
-					ch_in_word = 0;
-				}
-				ch_in_line++;
-				if (line[ch_in_line] == '\0') {
-					data[word][ch_in_word] = '\0';
-					word++;
-					ch_in_word = 0;
-				}
-			}
-		}
-		/* Convert the string data to hexadecimal */
-		addr = strtoul(data[0], NULL, 16);
-		next_addr = strtoul(data[3], NULL, 16);
-		if (search_addr >= addr && search_addr < next_addr) {
-			total = snprintf(buffer, buflen, "%s", data[2]);
-			sym_offset = search_addr - addr;
-			total_size = next_addr - addr;
-			snprintf(&buffer[total - 1], buflen - total, "+0x%lx/0x%lx", sym_offset, total_size);
-			break;
-		}
-		if (search_addr < addr) {
+	getsymbolnum();
```

Comment:
Please perform NULL check on buffer parameter before using it.

Reply:
Ok. Will modify accordingly.